soc/arm/nordic_nrf/nrf91/soc.c

```diff
@@ -68,5 +68,17 @@ void z_platform_init(void)
 	SystemInit();
 }
 
+#ifdef CONFIG_TRUSTED_EXECUTION_NONSECURE
+#include <secure_services.h>
+/* Overrides the weak ARM implementation:
+   Call into secure firmware if in non-secure. */
```

Comment:
Could you say why we would rather want to to do this than use the weak ARM implementation?

For instance, does the non-secure FW not have the rights to do a reboot, or does SPM just want
to be kept in the loop about reboots?

Reply:
Added a comment. I moved this to a separate file in https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/753